docs/REQUIREMENTS.md

```diff
@@ -0,0 +1,79 @@
+
+### Laclan
+
+- As a user, I want to store a signature/SBOM/etc in a registry along with a reference to the container image <by digest?> it is for
+- As a user, I want to query the registry the signatures for a container image by name and tag
+- As a user, I want to query the registry for all stored objects that reference a container image by name and tag
+- As a user, I want to query the registry for stored objects that reference a container image filtering by type (eg. Signature, SBOM, etc) or by annotation (I want to see all signatures from this identity)
+- As a user, I want to store a signature for an SBOM in a registry along with a reference to SBOM it is for
+- As a user, I want to store a signature for an SBOM in a registry along with a reference to SBOM it is for
+- As a user, I want to query the registry for the most recent stored objects that reference a container image
+- As a user, I want all stored objects that reference a container image to be deleted when a container image is deleted
+- As a user, I want to be able to store a SBOM or signature in a registry without the container image it’s for
+- As a user, I want to be able to copy a container image and all it’s references to another registry
+
+### Steve
+
+- As a user, I want to copy a container image, and a subset of its references to another registry
+- As a user, I want to view the top-level container images in a registry through tags, but only see the references as a selected detail
+- As a registry operator, I want to help users understand how they can manage the lifecycle of their container images and their references
+- As a registry operator, I want to provide tag locking features that don’t preclude users from adding signatures or SBOMs to a tagged artifact
+- As an artifact author, I want to use multiple blobs/layers for reference types
+- As an artifact author, I want to push annotations (without blob data) as additional metadata, as reference types
+- As a security admin, I want to assure non-container artifacts, pushed to a registry doesn't add security risks to container runtime services that may pull the reference type 
+- As a container image author, I want to support promotion of container images, and their references, across registries that may not yet support references
+- As a security scanner project/product, I want to know the type of each artifact so I know how to scan it for vulnerabilities, or if I should scan it. (example, scanners evaluate signatures, but don’t "scan" them)
+
+### Nisha
+
+- As a user, I want to identify the most updated artifact in a registry
+- As a user, I would like to be able to map monotonically increasing product versions to container images so I have an idea of deployment progression
+- As a developer/devopser, I would like to bisect builds based on container images I have deployed
+
+### Josh
+
+- As a user, I want to store images in one registry, and signatures/SBOMs/attestations in a separate registry
+
+### Brandon
```

Comment:
Channeling a BDFL that says something along the lines of "don't break user space", I think these got lost in the shuffle:

- As a user, I would like graceful degradation if the registry does not support the new ref types. This would either be some kind of limited fall back support or a clear error message from tooling. (Brandon)
- As a user, I want to be sure existing runtimes are not affected by any of the new reference types added to images. (Brandon)

Reply:
my apologies I missed those! and of course, we can continue to add more of these later